packages/vue/src/composables/useAuth.ts

```diff
@@ -1,43 +1,55 @@
+import { createSharedComposable } from '@vueuse/core';
+import { reactive, watchEffect, onScopeDispose } from 'vue';
 import { useActor } from '@xstate/vue';
-import { ref, reactive, Ref, watchEffect } from 'vue';
-import { getServiceFacade } from '@aws-amplify/ui';
+import { interpret } from 'xstate';
+
+import {
+  AuthInterpreter,
+  createAuthenticatorMachine,
+  defaultAuthHubHandler,
+  getServiceFacade,
+  listenToAuthHub,
+} from '@aws-amplify/ui';
+
+import { UseAuth } from '../types';
 import { facade } from './useUtils';
-import { InterpretService } from '@/components';
-
-const service = ref() as Ref<InterpretService>;
-const useAuthenticatorValue = reactive({
-  ...facade,
-  send: '' as unknown,
-  state: '' as unknown,
-}) as any;
-
-export const useAuth = (serv?: InterpretService) => {
-  if (serv) {
-    service.value = serv;
-  }
-  return useActor(service.value);
-};
-
-const useInternalAuthenticator = () => {
-  createValues();
-  watchEffect(() => {
-    createValues();
-  });
-  return useAuthenticatorValue;
-};
 
-export const useAuthenticator = useInternalAuthenticator;
+export const useAuth = createSharedComposable((): UseAuth => {
+  const machine = createAuthenticatorMachine();
```

Comment:
Unlike React hooks, this gets only run once because composable gets called in `setup` [[docs](https://vuejs.org/guide/reusability/composables.html#conventions-and-best-practices)]. 

Only the reactive part gets updated independently.